spring-kafka/src/main/java/org/springframework/kafka/listener/MessageListener.java

```diff
@@ -26,8 +28,13 @@
  *
  * @author Marius Bogoevici
  * @author Gary Russell
+ * @author Sanghyeok An
  */
 @FunctionalInterface
 public interface MessageListener<K, V> extends GenericMessageListener<ConsumerRecord<K, V>> {
 
+	default void setCallbackForAsyncFailure(
```

Comment:
As I mentioned in the container review, we might consider just have this contract on the `MessagingMessageListenerAdapter` and that's it.